tools/tgs4_scripts/PreCompile.bat

```diff
@@ -0,0 +1,5 @@
+@echo off
+set TG_BOOTSTRAP_CACHE=%cd%
+cd %1
+set TG_BUILD_TGS_MODE=1
+tools\build\build
```

Comment:
tgs does not log the output of these scripts, just their exit code, so you will need to redirect stderr/out to log(s)s in the data folder so we can debug these things.

Testing this on event-hall all I get is:

```
Preaction Event: precompile @ C:\TGSTATION-SERVER-3-EVENTS-US\EventHandlers/precompile.bat ran. Exit Code: 0
```

```
Compile error: 
DM compiler version 513.1535
loading tgstation.dme
loading interface\skin.dmf
code\modules\asset_cache\asset_list_items.dm:6:error: 'tgui/public/tgui-common.chunk.js': cannot find file
code\modules\asset_cache\asset_list_items.dm:12:error: 'tgui/public/tgui.bundle.js': cannot find file
code\modules\asset_cache\asset_list_items.dm:13:error: 'tgui/public/tgui.bundle.css': cannot find file
code\modules\asset_cache\asset_list_items.dm:19:error: 'tgui/public/tgui-panel.bundle.js': cannot find file
code\modules\asset_cache\asset_list_items.dm:20:error: 'tgui/public/tgui-panel.bundle.css': cannot find file
tgstation.dmb - 5 errors, 0 warnings (12/19/20 2:38 am)
Total time: 1:08

```

Reply:
@Cyberboss

Reply:
You fail me yet again TGS3. @SpaceManiac @spookydonut can one you take care of this, please?

Reply:
> tgs does not log the output of these scripts

annoying

> redirect stderr/out to log(s)s in the data folder

will EventHandlers/last-command.log suffice or do we definitely need this to be customizable? EventHandlers/ is also used as the cache directory so at minimum I assume it is not deleted

Reply:
> last-command.log is in whatever the working directory of PreCompile.bat is

correct

And given we just need to set it up once, yes it should suffice

Reply:
If I get that log I can try to figure out what went wrong, although my intuition is telling me that it's something dumb like a path with spaces in

Reply:
Looks like PS is too old on the machines to use `Expand-Archive`
![image](https://user-images.githubusercontent.com/8171642/102695938-ab813a80-41f8-11eb-8564-422d2d68bd20.png)

Try using `[System.IO.Compression.ZipFile]::ExtractToDirectory()`

Reply:
>last-command.log

Could this shit like be better fucking documented then. because I was looking in event viewers, you know, where tgs fucking puts other output of commands like compile errors.

@Cyberboss

Reply:
it's up to Cyberboss whether TGS4 will log this stuff in accordance with TGS's standards. The last-command.log is something new in this PR with the intent of making it easier to get build logs from normal contributors looking for support

Reply:
TGS4 will log stdout/stderr, which this does write to.